Project/Extensions/UIColor.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  UIColor.swift
+//  NetguruCollegeApp
+//
+import UIKit
+
+extension UIColor {
+    /// Creates an instance of UIColor from given hexadecimal number given as a String.
+    /// Fails if given parameter is inncorect hexadecimal number.
+    ///
+    /// - Parameters:
+    ///   - hexString: a hex representation of color. 3, 6 or 8 characters with or without a '#'
+    convenience init?(hexString: String) {
```

Comment:
Can you please explain this part? How did exactly you implement?
What do you think about this implementation:

```    /// Initializes and returns a color object using the hex value.
    /// Parameter hex: The hex value of the color object.
    /// Returns: The color object.
    static func with(hex: Int) -> UIColor {
        return color(red:(hex >> 16) & 0xff, green:(hex >> 8) & 0xff, blue:hex & 0xff)
    }


    /// Initializes and returns a color object using the specified opacity and RGB component values.
    /// - Parameters:
    ///     - red: The red value of the color object.
    ///     - green: The green value of the color object.
    ///     - blue: The blue value of the color object.
    ///     - alpha: The opacity value of the color object.
    /// Returns: The color object.
    static func color(red: Int, green: Int, blue: Int, alpha: Int = 0xFF) -> UIColor {
        assert(red >= 0 && red <= 255, "Invalid red component")
        assert(green >= 0 && green <= 255, "Invalid green component")
        assert(blue >= 0 && blue <= 255, "Invalid blue component")
        return UIColor(red: CGFloat(red) / 255.0, green: CGFloat(green) / 255.0, blue: CGFloat(blue) / 255.0, alpha: CGFloat(alpha) / 255.0)
    }
```